lib/phpSec/Store/Pdo.php

```diff
@@ -1,273 +1,280 @@
-    if($mac != $data['mac']) {
-      throw new \phpSec\Exception\GeneralSecurityException('Message authentication code invalid while reading store');
-      return false;
-    }
-
-    /* And success! */
-    return unserialize($data['data']);
-  }
-
-  public function write($type, $id, $data) {
-    $crypto = $this->psl['crypt/crypto'];
-
-    /* Delete existing data first, to prevent a huge database. */
-    $this->delete($type, $id);
-
-    /* Prepeare query. */
-    $sth = $this->dbh->prepare(
-      'INSERT INTO '.$this->table.' (`id`, `mac`, `time`, `type`, `data`)' .
-      'VALUES(:id, :mac, :time, :type, :data)'
-    );
-
-    /* Serialize data, and create a MAC. */
-    $data = serialize($data);
-    $mac  = $crypto->pbkdf2($data, $id, 1000, 32);
-
-    /* We use this array to say what data goes where in the query. */
-    $data = array(
-      'id'   => $id,
-      'mac'  => $mac,
-      'time' => time(),
-      'type' => $type,
-      'data' => $data,
-    );
-
-    /* And, insert. */
-    return $sth->execute($data);
-  }
-
-  public function delete($type, $id){
-    $sth = $this->dbh->prepare(
-      'DELETE FROM '.$this->table.' WHERE type = :type AND id = :id'
-    );
-
-    $data = array(
-      'id'   => $id,
-      'type' => $type,
-    );
-
-    return $sth->execute($data);
-  }
-
-  public function listIds($type) {
-    $ids = array();
-
-    $sth = $this->dbh->prepare(
-      'SELECT * FROM '.$this->table.' WHERE type = :type'
-    );
-
-    $data = array(
-      'type' => $type,
-    );
-
-    $sth->execute($data);
-
-    while($row = $sth->fetch(\PDO::FETCH_ASSOC)) {
-      $ids[] = $row['id'];
-    }
-
-    return $ids;
-  }
-
-  public function meta($type, $id) {
-    $sth = $this->dbh->prepare(
-      'SELECT * FROM '.$this->table.' WHERE type = :type AND id = :id LIMIT 1'
-    );
-
-    $data = array(
-      'id'   => $id,
-      'type' => $type,
-    );
-
-    $sth->execute($data);
-
-    $meta = $sth->fetch(\PDO::FETCH_ASSOC);
-    if($meta === false) {
-      return false;
-    }
-
-    $obj = new \StdClass;
-
-    $obj->id   = $meta['id'];
-    $obj->mac  = $meta['mac'];
-    $obj->time = $meta['time'];
-
-    return $obj;
-  }
-
-  /**
-   * Extract configuration variables from the DSN.
-   *
-   * @param string $dsn
-   *   DSN to extract variables from.
-   *
-   * @return array
-   *   Returns an array with variable names as keys with corresponding values.
-   */
-  private function parseDsn($dsn) {
-    $parsed = array();
-
-    $typeEnd = strpos($dsn, ':');
-    $dbType = substr($dsn, 0, $typeEnd);
-    $parsed['dbtype'] = $dbType;
-
-    $dsn = substr($dsn, $typeEnd+1);
-
-    $parts = explode(';', $dsn);
-    foreach($parts as $part) {
-      list($key, $val) = explode('=', $part);
-      $parsed[$key] = $val;
-    }
-
-    return $parsed;
-  }
-}
\ No newline at end of file
+  private $psl = null;
+
+  public function __construct($loc, \phpSec\Core $psl, \PDO $dbh = null, $table = null) {

```

Comment:
Why does $loc remain unused and does this constructor introduce a new parameter $table? 

Perhaps it's better to pass the table name through $loc as normal and make the third param in the constructor dual purpose (either additional connection options, or existing PDO instance). This would be more future-feature compatible.